### PR TITLE
feat: add login form

### PR DESCRIPTION
**Pull request by LBokan**
### Description

Added libraries and set them up: Vuetify and Vuelidate. Created login page layout. Set validation for fields

### Type of change

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [ ] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist

- [x] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)

> If not explain here

- [x] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change

Add screenshots of the changed components here.

#### Screenshots/Video after change

login-form-preview
login-form-validation1
login-form-validation2
login-form-validation3

https://github.com/LBokan/CarInsurance-Bootcamp/assets/117748123/9ca5e279-7d58-417d-8eb7-1141eac18107

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -12,9 +12,13 @@
     "format": "prettier --write src/"
   },
   "dependencies": {
+    "@mdi/font": "5.9.55",
+    "@vuelidate/core": "^2.0.2",
```
Comment (lavrovpy) [inline frontend/package.json:16]: is vuelidate really required here?

File: frontend/src/hooks/useValidateLoginForm.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { useVuelidate } from '@vuelidate/core';
+import { email, required, minLength } from '@vuelidate/validators';
```
Comment (lavrovpy) [inline frontend/src/hooks/useValidateLoginForm.ts:2]: Better to use computed properties for simple validation

File: frontend/src/hooks/useValidateLoginForm.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+} from '@vuelidate/core';
+
+interface ILoginState {
+  email: string,
+  password: string
+}
+
+interface ILoginRules {
+  email: {
+    required: ValidationRuleWithoutParams<any>,
```
Comment (lavrovpy) [inline frontend/src/hooks/useValidateLoginForm.ts:20]: Replace any with the actual type.
use _any_ should be the last resort option

File: frontend/src/hooks/useValidateLoginForm.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        required, minLength: minLength(8)
+    }
+  };
+
+  const v$ = useVuelidate( rules, state );
+
+  const getErrorMessage = computed<TypeGetErrorMessage>(() => {
+    return (stateVal: string) => {
+        return v$?.value?.[stateVal]?.$error ? 
+        (v$?.value?.[stateVal]?.$errors.map((e: ErrorObject) => e.$message) as string[]) :
```
Comment (lavrovpy) [inline frontend/src/hooks/useValidateLoginForm.ts:51]: This line seems unreadable. better to refactor it

File: frontend/package.json (hunk 1 above)
Comment (LBokan) [reply frontend/package.json]: Removed Vuelidate library

File: frontend/src/hooks/useValidateLoginForm.ts (hunk 3 above)
Comment (LBokan) [reply frontend/src/hooks/useValidateLoginForm.ts]: Removed these types (they were from Vuelidate lib.)

File: frontend/src/hooks/useValidateLoginForm.ts (hunk 2 above)
Comment (LBokan) [reply frontend/src/hooks/useValidateLoginForm.ts]: Done

File: frontend/src/hooks/useValidateLoginForm.ts (hunk 4 above)
Comment (LBokan) [reply frontend/src/hooks/useValidateLoginForm.ts]: Removed this func, it was connected with Vuelidate lib.

File: frontend/src/App.vue [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
\ No newline at end of file
+  <v-layout class="bg-blue-lighten-1">
+    <v-main class="app">
+      <router-view />
+    </v-main>
+  </v-layout>
+</template>
+
+<style>
+  * {
```
Comment (lavrovpy) [inline frontend/src/App.vue:10]: This is CSS reset stuff. Better to create a _reset.css_ file and import it in _main.ts_
you can use this one for example [https://gist.github.com/Asjas/4b0736108d56197fce0ec9068145b421](https://gist.github.com/Asjas/4b0736108d56197fce0ec9068145b421)

File: frontend/src/App.vue [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+  }
+
+  .app {
+    display: flex;
+    flex-direction: column;
+    height: 100vh;
```
Comment (lavrovpy) [inline frontend/src/App.vue:19]: usign vh units can cause issues on mobile devices. Consider using dvh. More details: https://www.youtube.com/watch?v=veEqYQlfNx8
https://www.youtube.com/watch?v=xl9R8aTOW_I

File: frontend/src/components/LoginForm.vue [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    <v-btn 
+      color="success"
+      size="x-large"
+      rounded="lg"
+      block
+      :disabled="isSubmitButtonDisabled"
+      type="submit"
+      variant="elevated"
+    >
+      LOG IN
```
Comment (lavrovpy) [inline frontend/src/components/LoginForm.vue:45]: Avoid hardcoding capital letters. Use text-tranform instead

**Review by lavrovpy — APPROVED** [review]
It's good to merge after solving the 3 comments

File: frontend/src/App.vue (hunk 5 above)
Comment (LBokan) [reply frontend/src/App.vue:10]: Deleted styles for "*", cause vuetify file with styles already has these styles

File: frontend/src/App.vue (hunk 6 above)
Comment (LBokan) [reply frontend/src/App.vue:19]: Done

File: frontend/src/components/LoginForm.vue (hunk 7 above)
Comment (LBokan) [reply frontend/src/components/LoginForm.vue]: Done

**Pull request merged**